Assets/MixedRealityToolkit.Tests/EditModeTests/Core/Utilities/PackageManifestUpdaterTests.cs

```diff
@@ -0,0 +1,164 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Utilities.Editor;
+using NUnit.Framework;
+using System;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Tests.EditMode.Core.Utilities.Editor
+{
+    /// <summary>
+    /// Tests that verify the correct behavior of the PackageManifestUpdater.
+    /// </summary>
+    public class PackageManifestUpdaterTests
+    {
+        /// <summary>
+        /// Verifies parsing of a properly formatted value returns the correct results.
+        /// </summary>
+        [Test]
+        public void TryGetValidVersion()
+        {
+            Debug.Log("Try to get version from properly formatted version with no prerelease.");
+
+            Version v;
+            float f;
+
+            bool success = PackageManifestUpdater.TryGetVersionComponents("17.27.43", out v, out f);
+            Assert.IsTrue(success);
+            Assert.AreEqual(v, new Version(17, 27, 43));
+            Assert.AreEqual(f, 0f);
+
+            Debug.Log("Try to get version from properly formatted version with prerelease.");
+
+            success = PackageManifestUpdater.TryGetVersionComponents("0.9.1-20200131.12", out v, out f);
+            Assert.IsTrue(success);
+            Assert.AreEqual(v, new Version(0, 9, 1));
+            Assert.AreEqual(f, float.Parse("20200131.12"));
+        }
+
+        /// <summary>
+        /// Verifies parsing of an improperly formatted string returns the correct failure results.
+        /// </summary>
+        [Test]
+        public void TryGetInvalidVersion()
+        {
+            Debug.Log("Try to get version from improperly formatted version.");
+            Version v;
+            float f;
+
+            bool success = PackageManifestUpdater.TryGetVersionComponents("x.2.3", out v, out f);
+            Assert.IsFalse(success);
+            Assert.IsNull(v);
+            Assert.AreEqual(f, float.NaN);
+
+            Debug.Log("Try to get version from improperly formatted prerelease.");
+
+            v = new Version(5, 6, 7);
```

Comment:
If this is to test that v and f are set to null/NaN Values, it might be helpful to leave a comment saying that these are being set to arbitrary values and the specific values don't actually matter.

Reply:
sounds good

Reply:
done